src/setupWorkerModule.ts

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License.
+
+import { HookCallback } from '@azure/functions-worker';
+import { Disposable } from './Disposable';
+import { WorkerChannel } from './WorkerChannel';
+import Module = require('module');
+
+export function setupWorkerModule(channel: WorkerChannel): void {
```

Comment:
Is this code needed to provide `registerHook` from build-in `@azure/functions-worker` without specifying `@azure/functions-worker` in the function app `package.json`?

If so can you please add a comment describing this.

Reply:
> Is this code needed to provide registerHook from build-in @azure/functions-worker without specifying @azure/functions-worker in the function app package.json?

Correct

> when require('@azure/functions-worker') will be called? in sample usage there is:
import { registerHook } from '@azure/functions-worker';

The sample usage is in typescript. When you compile `import` to javascript, it gets converted to `require`

> Can you please add a comment to describe what the function is doing.

Done

Reply:
A follow-up question to this: does this mean `@azure/functions-worker` can't be used with ESM?

Reply:
It looks a little different, but yes it can be used with ESM

```
import { createRequire } from "module";
const require = createRequire(import.meta.url);
const { registerHook } = require('@azure/functions-core');

registerHook('preInvocation', async (context) => {
    context.invocationContext.log('HELLO FROM PRE INVOCATION mjs');
});
```

This is the standard pattern for using a `require` method within ESM

Reply:
You can't use a simple `import`, but frankly I don't think that problem is unique to us. When I originally looked into this, the closest I could find to a solution is a preview "loaders" feature which may or may not help us in the future:
- https://github.com/nodejs/loaders
- https://nodejs.org/api/esm.html#loaders

I think there's plenty of oddities with Node's switch to ESM that are still being ironed out, so I think people using ESM should be fine with a few extra lines of code as shown above.

Reply:
Makes sense, thanks for looking into it!